app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -45,7 +45,7 @@ def instances
       # saving hardware information (CPU, Memory, etc.)
       ps_hw = persister.hardwares.build(
         :vm_or_template  => ps_vmi,
-        :cpu_total_cores => Float(instance["processors"]).ceil,
+        :cpu_total_cores => Integer(instance['virtualCores']['assigned']),
```

Comment:
`instance['virtualCores']['assigned']` is already an Integer type so there's no need to pass it to `Integer()`:
```
irb(main):039:0> instance['virtualCores']['assigned'].kind_of? Integer
=> true
```

We can do this in another commit, but the same goes for the next line with `instance["memory"]`.